File: uri/ad-hock/snakes_and_ladders/1569.cpp
#include <iostream>
#include <map>
#include <vector>

using namespace std;

int main() {
    int t;
    cin >> t;
    
    while (t--) {
        int a, b, c;
        cin >> a >> b >> c;
        
        map<int, int> board;
        
        for (int i = 0; i < b; i++) {
            int from, to;
            cin >> from >> to;
            board[from] = to;
        }
        
        vector<int> players(a + 1, 1);
        
        bool gameOver = false;
        int currentPlayer = 1;
        
        for (int i = 0; i < c; i++) {
            int roll;
            cin >> roll;
            
            if (!gameOver) {
                players[currentPlayer] += roll;
                
                if (players[currentPlayer] > 100) {
                    players[currentPlayer] = 100;
                }
                
                if (board.count(players[currentPlayer])) {
                    players[currentPlayer] = board[players[currentPlayer]];
                }
                
                if (players[currentPlayer] == 100) {
                    gameOver = true;
                }
                
                currentPlayer++;
                if (currentPlayer > a) {
                    currentPlayer = 1;
                }
            }
        }
        
        for (int i = 1; i <= a; i++) {
            cout << "Position of player " << i << " is " << players[i] << "." << endl;
        }
    }
    
    return 0;
}

File: uri/ad-hock/snakes_and_ladders/1569.rs
use std::io;
use std::collections::HashMap;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();
    
    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let nums: Vec<usize> = input.trim().split_whitespace().map(|s| s.parse().unwrap()).collect();
        let a = nums[0];
        let b = nums[1];
        let c = nums[2];
        
        let mut board: HashMap<i32, i32> = HashMap::new();
        
        for _ in 0..b {
            input.clear();
            io::stdin().read_line(&mut input).unwrap();
            let parts: Vec<i32> = input.trim().split_whitespace().map(|s| s.parse().unwrap()).collect();
            board.insert(parts[0], parts[1]);
        }
        
        let mut players = vec![1; a + 1];
        
        let mut game_over = false;
        let mut current_player = 1;
        
        for _ in 0..c {
            input.clear();
            io::stdin().read_line(&mut input).unwrap();
            let roll: i32 = input.trim().parse().unwrap();
            
            if !game_over {
                players[current_player] += roll;
                
                if players[current_player] > 100 {
                    players[current_player] = 100;
                }
                
                if let Some(&destination) = board.get(&players[current_player]) {
                    players[current_player] = destination;
                }
                
                if players[current_player] == 100 {
                    game_over = true;
                }
                
                current_player += 1;
                if current_player > a {
                    current_player = 1;
                }
            }
        }
        
        for i in 1..=a {
            println!("Position of player {} is {}.", i, players[i]);
        }
    }
}
